examples/stable_diffusion_v2/ldm/data/dataset.py

```diff
@@ -276,15 +276,18 @@ class BatchSampler:
     Batch Sampler
     """
 
-    def __init__(self, lens, batch_size, device_num):
+    def __init__(self, lens, batch_size, device_num, shuffle):
         self._lens = lens
         self._batch_size = batch_size * device_num
+        self._shuffle = shuffle
 
     def _create_ids(self):
         return list(range(self._lens))
 
     def __iter__(self):
         ids = self._create_ids()
+        if self._shuffle:
+            shuffle(ids)
```

Comment:
Why shuffle in each batch iteration?  Instead, it is expected to shuffle the indices each epoch.